lib/usd/translators/meshWriter_BlendShapes.cpp

```diff
@@ -0,0 +1,931 @@
+    assert(static_cast<int>(numInputTargetGrps) > targetGroupIndex);
+    MPlug plgInputTargetGrp = plgInputTargetGrps.elementByPhysicalIndex(targetGroupIndex, &stat);
+    CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+    MPlug plgInputTargetItems = mayaFindChildPlugWithName(plgInputTargetGrp, "inputTargetItem");
+    assert(!plgInputTargetItems.isNull());
+    assert(plgInputTargetItems.isArray());
+
+#if _DEBUG // TODO: (yliangsiew) Need to find a good macro that I can rely on for this.
+    MIntArray plgInputTargetItemLogicalIndices;
+    plgInputTargetItems.getExistingArrayAttributeIndices(plgInputTargetItemLogicalIndices, &stat);
+    CHECK_MSTATUS_AND_RETURN_IT(stat);
+    assert(mayaSearchMIntArray(weightIndex, plgInputTargetItemLogicalIndices) == true);
+#endif
+
+    MPlug plgInputTargetItem = plgInputTargetItems.elementByLogicalIndex(targetWeightIndex, &stat);
+    CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+    MPlug plgInputComponentsTarget
+        = mayaFindChildPlugWithName(plgInputTargetItem, "inputComponentsTarget");
+    assert(!plgInputComponentsTarget.isNull());
+
+    // NOTE: (yliangsiew) Problem: looks like there's a maya bug where you have to twiddle the
+    // blendshape weight directly before these kComponentListData-type plugs get evaluated.
+#if MAYA_BLENDSHAPE_EVAL_HOTFIX
+    MPlug plgBlendShapeWeights = fnNode.findPlug("weight", false, &stat);
+    CHECK_MSTATUS_AND_RETURN_IT(stat);
+    assert(plgBlendShapeWeights.isArray());
+
+    MFnBlendShapeDeformer fnBS(blendShapeNode, &stat);
+    MFloatArray           origWeightVals; // Storage for the original weight values.
+    MIntArray             indicesChanged;
+    MIntArray             targetItemIndexList;
+    MIntArray             bsWeightIndices;
+    stat = fnBS.weightIndexList(bsWeightIndices);
+    CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+    // NOTE: (yliangsiew) Need to force the other kComponentListData plugs to
+    // be dirtied and recalculated.
+    MObjectArray baseObjs;
+    stat = fnBS.getBaseObjects(baseObjs);
+    CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+    for (unsigned int i = 0; i < bsWeightIndices.length(); ++i) {
+        int weightIndex = bsWeightIndices[i];
+        for (unsigned int j = 0; j < baseObjs.length(); ++j) {
+            MObject baseObj = baseObjs[j];
+            targetItemIndexList.clear();
+            stat = fnBS.targetItemIndexList(weightIndex, baseObj, targetItemIndexList);
+            CHECK_MSTATUS_AND_RETURN_IT(stat);
+
+            for (unsigned int k = 0; k < targetItemIndexList.length(); ++k) {
+                float curWeight = fnBS.weight(weightIndex, &stat);
+                CHECK_MSTATUS_AND_RETURN_IT(stat);
+                origWeightVals.append(curWeight);
+                // NOTE: (yliangsiew) For in-between shapes, need to trigger at _all_
+                // full weight values of each target so as to populate the components
+                // list for each of the targets. Yea, this is dumb.
```

Comment:
I see it is even more annoying that I realized. The blend shape is doing lazy evaluation and only computing input target information for targets it needs, so in the in-between case there will be lots of targets that don't get evaluated.